packages/cli/src/services/import.service.ts

```diff
@@ -94,6 +117,266 @@ export class ImportService {
+	}
+
+	async importEntities(inputDir: string, truncateTables: boolean) {
+		validateDbTypeForImportEntities(this.dataSource.options.type);
+
+		await this.dataSource.transaction(async (transactionManager: EntityManager) => {
+			await this.disableForeignKeyConstraints(transactionManager);
+
+			// Get import metadata after migration validation
+			const importMetadata = await this.getImportMetadata(inputDir);
+			const { tableNames, entityFiles } = importMetadata;
+			const entityNames = Object.keys(entityFiles);
+
+			if (truncateTables) {
+				this.logger.info('\n🗑️  Truncating tables before import...');
+
+				this.logger.info(`Found ${tableNames.length} tables to truncate: ${tableNames.join(', ')}`);
+
+				await Promise.all(
+					tableNames.map(
+						async (tableName) => await this.truncateEntityTable(tableName, transactionManager),
+					),
+				);
+
+				this.logger.info('✅ All tables truncated successfully');
+			}
+
+			if (!truncateTables && !(await this.areAllEntityTablesEmpty(tableNames))) {
+				this.logger.info(
+					'\n🗑️  Not all tables are empty, skipping import, you can use --truncateTables to truncate tables before import if needed',
+				);
+				return;
+			}
+
+			// Import entities from the specified directory
+			await this.importEntitiesFromFiles(inputDir, transactionManager, entityNames, entityFiles);
+
+			await this.enableForeignKeyConstraints(transactionManager);
+		});
+	}
+
+	/**
+	 * Import entities from JSONL files into the database
+	 * @param inputDir - Directory containing exported entity files
+	 * @param transactionManager - TypeORM transaction manager
+	 * @param entityNames - Array of entity names to import
+	 * @param entityFiles - Record of entity names to their file paths
+	 * @returns Promise that resolves when all entities are imported
+	 */
+	async importEntitiesFromFiles(
+		inputDir: string,
+		transactionManager: EntityManager,
+		entityNames: string[],
+		entityFiles: Record<string, string[]>,
+	): Promise<void> {
+		this.logger.info(`\n🚀 Starting entity import from directory: ${inputDir}`);
+
+		if (entityNames.length === 0) {
+			this.logger.warn('No entity files found in input directory');
+			return;
+		}
+
+		this.logger.info(`📋 Found ${entityNames.length} entity types to import:`);
+
+		let totalEntitiesImported = 0;
+
+		await Promise.all(
+			entityNames.map(async (entityName) => {
+				const files = entityFiles[entityName];
+				this.logger.info(`   • ${entityName}: ${files.length} file(s)`);
+				this.logger.info(`\n📊 Importing ${entityName} entities...`);
+
+				const entityMetadata = this.dataSource.entityMetadatas.find(
+					(meta) => meta.name.toLowerCase() === entityName,
+				);
+
+				if (!entityMetadata) {
+					this.logger.warn(`   ⚠️  No entity metadata found for ${entityName}, skipping...`);
+					return;
+				}
+
+				const tableName = entityMetadata.tableName;
+				this.logger.info(`   📋 Target table: ${tableName}`);
+
+				let entityCount = 0;
+				await Promise.all(
```

Comment:
I'm a bit worried about the promise.all inside a promise.all, in terms of either simultaneous queries to the db or io reading. But I'm not sure about the expected volumetry